### ISSUE TITLE
"Connecting to React…" forever in FireFox with `tiny-lr` LiveReload server

### ISSUE THREAD

**Issue by dahjelle** [facebook/react-devtools#869]
**Browser**: Firefox Nightly 57.0a1 (2017-08-11)
**DevTools**: 2.5.0

I have a project where I'm using the [`tiny-lr`](https://www.npmjs.com/package/tiny-lr) LiveReload server. On Firefox Nightly, the DevTools perpetually display "Connecting to React…" and never load. It works fine in Chrome.

If I turn the LiveReload server *off* , the DevTools work without a problem, but, of course, I lose the LiveReload functionality. Interestingly, I can even kill the LiveReload server while waiting for the DevTools to load, and they will suddenly finish loading properly.

**Comment by dahjelle** [comment]
I tried making a minimal test case using `tiny-lr` in an app made with `create-react-app`, but things seemed to work fine there. I'm not sure what else could be causing the issue…

**Comment by clarkbw** [comment]
Any chance you can put the existing project online somewhere?

**Comment by dahjelle** [comment]
> Any chance you can put the existing project online somewhere?

Unfortunately, no. As I've experimented more, it could also be related to a PouchDB/CouchDB `_changes` feed, as well…but I cannot reproduce that in a minimized test case, either.

Does the React DevTools wait for a page loaded event or anything of that sort that might be waiting for a long poll HTTP feed or a web socket to finish before loading the DevTools?

**Comment by dahjelle** [comment]
I just noticed another odd thing—sometimes, the DevTools eventually seem to come up. I haven't been able to figure out how to reproduce this, but sometimes I'll come back to Firefox and see that the React DevTools successfully loaded.

**Comment by dahjelle** [comment]
I've resolved this in the case of our app. The issue appears to have been having a PouchDB `_changes` feed subscription immediately in a `<script>` tag at the end of the document. This apparently causes Firefox to think it is perpetually in a loading state, with the spinner and everything. If I subscribed to the `_changes` feed in a `setTimeout`, then Firefox didn't think it was loading *and* the React DevTools load correctly.

I think my trouble reproducing it was subscribing to the `_changes` too late during page load.

Thanks!
